### ISSUE TITLE
Add support for instrumented executor service

### ISSUE THREAD

**Issue by dennyac** [dropwizard/dropwizard#2640]
###### Problem:
I'm attempting to improve visibility of [InstrumentedExecutorService](https://github.com/dropwizard/metrics/blob/4.1-development/metrics-core/src/main/java/com/codahale/metrics/InstrumentedExecutorService.java) / [InstrumentedScheduledExecutorService](https://github.com/dropwizard/metrics/blob/4.1-development/metrics-core/src/main/java/com/codahale/metrics/InstrumentedScheduledExecutorService.java) which is available in dropwizard metrics. I've worked with the dropwizard/metrics framework for close to four years now, and I wasn't aware that they existed. Hoping this helps others.

###### Solution:
I'm suggesting adding a method to the builders to provide the metricRegistry. If the metricRegistry is provided, it will return the instrumented executor service. 
```java
ExecutorService executorService = environment.lifecycle()
    .executorService(nameFormat)
    .maxThreads(maxThreads)
    .metricRegistry(environment.metrics()) // Returns InstrumentedExecutorService if metricRegistry is passed
    .build();

ScheduledExecutorService scheduledExecutorService = environment.lifecycle()
    .scheduledExecutorService(nameFormat)
    .metricRegistry(environment.metrics()) // Returns InstrumentedScheduledExecutorService if metricRegistry is passed
    .build();
```
I've added some documentation as well.

**Comment by dennyac** [comment]
PR which addresses this - https://github.com/dropwizard/dropwizard/pull/2639

**Comment by dennyac** [comment]
Based on suggestions I closed #2639 and raised two PRs which uses the the internal metric registry out of the box

I prefer #2650 over #2649  as it has more metrics, but I'll leave that decision to you all.

#2650 
### Pros: 
- More metrics - submitted(meter), running(counter), completed(meter), idle(timer), duration(timer)
### Cons:
- Developers won't have access to the extra methods ThreadPoolFactory and ScheduledThreadPoolFactory provide by type casting

#2650 
### Pros: 
- Developers can access the extra methods ThreadPoolFactory and ScheduledThreadPoolFactory provide by type casting
### Cons:
- Less metrics - created(meter), running(counter), terminated(meter)
